homeassistant/components/ecobee/climate.py

```diff
@@ -84,13 +93,37 @@
     PRESET_HOLD_INDEFINITE: "indefinite",
 }
 
-SERVICE_SET_FAN_MIN_ON_TIME = "set_fan_min_on_time"
+SERVICE_CREATE_VACATION = "create_vacation"
+SERVICE_DELETE_VACATION = "delete_vacation"
 SERVICE_RESUME_PROGRAM = "resume_program"
+SERVICE_SET_FAN_MIN_ON_TIME = "set_fan_min_on_time"
 
-SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema(
+DTGROUP_INCLUSIVE_MSG = (
+    f"{ATTR_START_DATE}, {ATTR_START_TIME}, {ATTR_END_DATE}, "
+    f"and {ATTR_END_TIME} must be specified together"
+)
+
+CREATE_VACATION_SCHEMA = vol.Schema(
     {
-        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
-        vol.Required(ATTR_FAN_MIN_ON_TIME): vol.Coerce(int),
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_VACATION_NAME): cv.string,
+        vol.Required(ATTR_COOL_TEMP): vol.Coerce(float),
+        vol.Required(ATTR_HEAT_TEMP): vol.Coerce(float),
+        vol.Inclusive(ATTR_START_DATE, "dtgroup", msg=DTGROUP_INCLUSIVE_MSG): cv.string,
```

Comment:
Maybe we should have validated the date and time as date and time? Future improvement possible. Sorry for missing this.

Reply:
I had thought of that - but it's really a string that gets passed to the ecobee API so we wouldn't necessarily want to convert it to a datetime object - and in any event if it's not in a valid format, then the vacation won't get created by the API.

Reply:
It would just be for validation. We like to validate as early as possible, ie in the service schema.

Reply:
Would this validate or transform?
```python
        vol.Inclusive(ATTR_START_DATE, "dtgroup", msg=DTGROUP_INCLUSIVE_MSG): cv.date,
        vol.Inclusive(ATTR_START_TIME, "dtgroup", msg=DTGROUP_INCLUSIVE_MSG): cv.time,
        vol.Inclusive(ATTR_END_DATE, "dtgroup", msg=DTGROUP_INCLUSIVE_MSG): cv.date,
        vol.Inclusive(ATTR_END_TIME, "dtgroup", msg=DTGROUP_INCLUSIVE_MSG): cv.time,
```

I'm happy to open a PR now to patch this.

Reply:
It would transform, so we'd need to transform back to string with another validator after the first one. Combine with `vol.All`.